### ISSUE TITLE
remove_watcher doesn't work anymore after GDPR username deprecation

### ISSUE THREAD

**Issue by doron-cohen** [pycontribs/jira#906]
Before raising any bug be sure that you used the latest release and that it still reproduces with the latest master. No other version is supported.

**Describe the bug**

`remove_watcher` uses the username but it's deprecated:

**To Reproduce**
```
use remove_watcher
```
1. Any additional steps or considerations that happen before or after.

**Expected behavior**

**Stack Trace**
```
  /home/xxxxx/lib/python3.4/site-packages/jira/client.py:108 .................... wrapper >> result = func(*arg_list, **kwargs)
  /home/xxxxx/lib/python3.4/site-packages/jira/client.py:1317 ............ remove_watcher >> result = self._session.delete(url, params=params)
  /home/xxxxx/lib/python3.4/site-packages/jira/resilientsession.py:159 ........... delete >> return self.__verb('DELETE', url, **kwargs)
  /home/xxxxx/lib/python3.4/site-packages/jira/resilientsession.py:146 ........... __verb >> raise_on_error(response, verb=verb, **kwargs)
  /home/xxxxx/lib/python3.4/site-packages/jira/resilientsession.py:56 .... raise_on_error >> r.status_code, error, r.url, request=request, response=r, **kwargs)
jira.exceptions.JIRAError: JiraError HTTP 401 url: https://xxxxxx.atlassian.net/rest/api/2/issue/REG-268153/watchers?username=someuser
	text: {"errorMessages":[],"errors":{}}
```

**Version Information**
Python Interpreter: 3.4.3
jira-python: 1.0.10 but all are affected
OS: linux
IPython (Optional): <VERSION>
Other Dependencies:

**Additional context**

**Comment by makegofast** [comment]
I also have run into this with v 2.0.0

**Comment by DoronCohen** [comment]
Fix should be easy. `accountId` should be passed instead of `username`. I will try to send a PR

**Comment by avallesalas** [comment]
Tested on my setup, changed client.py and resources.py from `username` to `accountId` and now works like a charm

**Comment by DoronCohen** [comment]
Yes thats how we patched this issue. I didnt have time for a PR

**Comment by adehad** [comment]
fixed in `pip install jira==3.1.0rc1` via #1109
